### ISSUE TITLE
Can't query experiments

### ISSUE THREAD

**Issue by AlFontal** [cytoflow/cytoflow#325]
### PLATFORM:

Linux

### OPERATING SYSTEM:

Ubuntu 20.04

### SEVERITY:

Minor

### DESCRIPTION: 

I don't fully understand the nature of the error, but I seem unable to directly query any experiment object. 

Minimal example with a Range2D gate I define to filter out samples:

```python

op_1 = flow.Range2DOp(name='DNA_positive',
                      xchannel='Ir191Di',
                      xlow=320.15367600329364,
                      xhigh=7055.0198037746595,
                      ychannel='Ir193Di',
                      ylow=315.44694493739735,
                      yhigh=14509.031608141782)

ex_1 = op_1.apply(ex_0)

ex_1.query(expr='DNA_positive == True')

```
This generates an `IndexError`: 
```
~/.cache/pypoetry/virtualenvs/pm-graphene-cytof-l3qb9n-H-py3.8/lib/python3.8/site-packages/cytoflow/experiment.py in query(self, expr, **kwargs)
    271             new_name = util.sanitize_identifier(name)
    272             if new_name in resolvers:
--> 273                 raise util.CytoflowError("Tried to sanitize column name {1} to "
    274                                          "{2} but it already existed in the "
    275                                          " DataFrame."

IndexError: Replacement index 2 out of range for positional args tuple
```
The error doesn't come from the query expression, since I can apply the same query to the experiment data directly with `pandas` with no issue:

```python
ex_1.data.query(expr='DNA_positive == True')
```
![image](https://user-images.githubusercontent.com/10958332/143481755-976f71df-8099-41ab-89b1-5a61e451f188.png)

The issue doesn't come either from the column itself, as I can't query or pass any expression at all, I always get the same error:
![image](https://user-images.githubusercontent.com/10958332/143481854-dd52a466-dcd1-4983-80d4-36c4fb01ee9a.png)
Which again works when querying the `pandas` dataframe itself:

![image](https://user-images.githubusercontent.com/10958332/143482051-feb8285e-fcf8-48b7-8475-7951b83cc684.png)

Even when using a nonsense string as query:

![image](https://user-images.githubusercontent.com/10958332/143481927-6b574f85-615c-4b70-9c7d-872f76d523e4.png)

Any ideas on where I might be messing up?

**Comment by kristjaneerik** [comment]
Could you list all the columns in the underlying dataframe, e.g. by doing
```python
print(ex.data.columns)
```

There are two things going on here. The error message you are seeing is due to a bug.
```python
                raise util.CytoflowError("Tried to sanitize column name {1} to "
                                         "{2} but it already existed in the "
                                         " DataFrame."
                                         .format(name, new_name))
```
should read
```python
                raise util.CytoflowError("Tried to sanitize column name {0} to "
                                         "{1} but it already existed in the "
                                         " DataFrame."
                                         .format(name, new_name))
```

But this piece of code is triggered by another issue, which is that two columns seem to be "sanitizing" to the same name. E.g. if you have a column "DNA positive" and then try to add a filter with the name "DNA_positive", they are distinct, but end up being sanitized to the same thing.

**Comment by AlFontal** [comment]
Hi @kristjaneerik! 

Thanks, you were right, I messed up the channel naming when instantiating the experiment and two different channels had the same name. After fixing that, I can query now without issues. However, I think that:

+ We should fix the issue with the `.format` indexes that you specified, do you want to open a PR?
+ Maybe there should be a check at experiment definition that makes sure no two channels share the same name (or resolve to the same name after 'sanization'?)

**Comment by bpteague** [comment]
> * We should fix the issue with the `.format` indexes that you specified, do you want to open a PR?

Yep, that's a bug. I'll fix it.

> * Maybe there should be a check at experiment definition that makes sure no two channels share the same name (or resolve to the same name after 'sanization'?)

Agreed. I'll add a test and a fix.

**Comment by bpteague** [comment]
Fixed in fa4f5934996187e57616bd396eeadc2d0e79a545
